### PR TITLE
Grafana Tempo

### PR DESCRIPTION
**Pull request by stephanos**
## What changed?

Added Grafana Tempo - an OTEL collector - to the developer setup.

Since we're already using Grafana, Tempo seemed like a logical choice.

## Why?

It allows developers to see all the RPC calls they are making (e.g. when running the server locally or a functional test).

Tempo allows filtering by workflow ID. _(note that not _all_ places are tagged yet, but lots of them)_

## How did you test it?

```
make start-grafana-tempo
make OTEL=true start
```
```
tctl --ns default namespace register
tctl wf start --tq test --wid test -wt test
```

image

And running a functional test from the IDE:

image

## Documentation

Updated docs/development/tracing.md

### CODE REVIEW

File: develop/docker-compose/docker-compose.jaeger.yml
Comment (stephanos) [inline develop/docker-compose/docker-compose.jaeger.yml:1]: Tempo can do everything Jaeger does, and more.

File: common/telemetry/env.go [hunk 1]
```diff
@@ -73,6 +73,8 @@ func SpanExportersFromEnv(
 
 			// other OTEL configuration env variables are picked up automatically by the exporter itself
 			exporters[OtelTracesOtlpExporterType] = otlptracegrpc.NewUnstarted()
+		case "none":
+			// ignored
```
Comment (stephanos) [inline common/telemetry/env.go:77]: A valid exporter value: https://opentelemetry.io/docs/specs/otel/configuration/sdk-environment-variables/#exporter-selection

File: tests/testcore/onebox.go
Comment (stephanos) [inline tests/testcore/onebox.go:1]: Allow functional tests to be configured by the same OTEL env vars.

Comment (stephanos) [inline tests/testcore/onebox.go:1]: Allow functional tests to be configured by the same OTEL env vars.

File: develop/docker-compose/docker-compose.jaeger.yml
Comment (stephanos) [inline develop/docker-compose/docker-compose.jaeger.yml:1]: Tempo can do everything Jaeger does, and more.

File: Makefile [hunk 2]
```diff
@@ -74,6 +74,14 @@ STAMPDIR := .stamp
 export PATH := $(ROOT)/$(LOCALBIN):$(PATH)
 GOINSTALL := GOBIN=$(ROOT)/$(LOCALBIN) go install
 
+OTEL ?= false
+ifeq ($(OTEL),true)
+	export OTEL_BSP_SCHEDULE_DELAY=0
```
Comment (stephanos) [inline Makefile:79]: Speeds up trace delivery.

**Comment by stephanos** [comment]
When running it in a functional test, it now also covers History and Matching:

Screenshot 2024-12-19 at 3 33 07 PM

File: temporal/fx.go [hunk 3]
```diff
@@ -859,6 +859,12 @@ func verifyPersistenceCompatibleVersion(config config.Persistence, persistenceSe
 	return nil
 }
 
+type TraceExportModuleInputs struct {
+	fx.In
+	Lifecycyle fx.Lifecycle
+	Config     *config.Config `optional:"true"`
```
Comment (stephanos) [inline temporal/fx.go:865]: Matching and History in functional tests don't have a config. Apparently.

**Review by alexshtin — APPROVED** [review]

**Comment by alexshtin** [comment]
It is really cool stuff. I am looking forward to try it for local development.

**Pull request merged**